`parquet / Build wasm32 (pull_request)` CI check failing on main

**Describe the bug**
`parquet / Build wasm32 (pull_request)` CI check is failing on main , evenwi

**To Reproduce**
Make a PR with no code changes, for example https://github.com/apache/arrow-rs/pull/5564

It fails like https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564

```
Run cargo build -p parquet --target wasm32-unknown-unknown
  cargo build -p parquet --target wasm3[2](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:2)-unknown-unknown
  shell: sh -e {0}
  env:
    RUSTFLAGS: -C debuginfo=1
    RUST_BACKTRACE: 1
   Compiling autocfg v1.2.0
   Compiling libm v0.2.8
   Compiling libc v0.2.15[3](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:3)
...
Compiling arrow-cast v51.0.0 (/__w/arrow-rs/arrow-rs/arrow-cast)
The following warnings were emitted during compilation:

warning: zstd-sys@2.0.10+zstd.1.5.6: In file included from zstd/lib/common/zstd_common.c:18:
warning: zstd-sys@2.0.10+zstd.1.5.6: In file included from zstd/lib/common/zstd_internal.h:35:
warning: zstd-sys@2.0.10+zstd.1.5.6: zstd/lib/common/xxhash.h:3157:22: error: called object type 'int' is not a function or function pointer
warning: zstd-sys@2.0.10+zstd.1.5.6:     XXH_STATIC_ASSERT(sizeof(XXH32_canonical_t) == sizeof(XXH32_hash_t));
warning: zstd-sys@2.0.10+zstd.1.5.6:     ~~~~~~~~~~~~~~~~~^
warning: zstd-sys@2.0.10+zstd.1.5.6: zstd/lib/common/xxhash.h:3600:22: error: called object type 'int' is not a function or function pointer
warning: zstd-sys@2.0.10+zstd.1.5.6:     XXH_STATIC_ASSERT(sizeof(XXH[64](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:65)_canonical_t) == sizeof(XXH64_hash_t));
warning: zstd-sys@2.0.10+zstd.1.5.6:     ~~~~~~~~~~~~~~~~~^
warning: zstd-sys@2.0.10+zstd.1.5.6: 2 errors generated.
...
```


**Expected behavior**
CI checks should pass when no code is chanegd


**Additional context**
Reported by @XiangpengHao  on https://github.com/apache/arrow-rs/pull/5564

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -53,6 +53,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"], option
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"], optional = true }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"], optional = true }
 zstd = { version = "0.13.0", optional = true, default-features = false }
+zstd-sys = { version = "2.0.9", optional = true, default-features = false }
 chrono = { workspace = true }
 num = { version = "0.4", default-features = false }
 num-bigint = { version = "0.4", default-features = false }
@@ -77,6 +78,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"] }
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"] }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"] }
 zstd = { version = "0.13", default-features = false }
+zstd-sys = { version = "2.0.9", default-features = false }
 serde_json = { version = "1.0", features = ["std"], default-features = false }
 arrow = { workspace = true, features = ["ipc", "test_utils", "prettyprint", "json"] }
 tokio = { version = "1.0", default-features = false, features = ["macros", "rt", "io-util", "fs"] }
@@ -87,7 +89,7 @@ object_store = { version = "0.9.0", default-features = false, features = ["azure
 all-features = true
 
 [features]
-default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "base64"]
+default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "zstd-sys", "base64"]
 # Enable lz4
 lz4 = ["lz4_flex"]
 # Enable arrow reader/writer APIs

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -55,7 +55,8 @@ lz4_flex = { version = "0.11", default-features = false, features = ["std", "fra
 zstd = { version = "0.13.0", optional = true, default-features = false }
 # TODO: temporary to fix parquet wasm build
 # upstream issue: https://github.com/gyscos/zstd-rs/issues/269
-zstd-sys = { version = "=2.0.9", optional = true, default-features = false }chrono = { workspace = true }
+zstd-sys = { version = "=2.0.9", optional = true, default-features = false }
+chrono = { workspace = true }
 num = { version = "0.4", default-features = false }
 num-bigint = { version = "0.4", default-features = false }
 base64 = { version = "0.22", default-features = false, features = ["std", ], optional = true }

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -53,8 +53,9 @@ brotli = { version = "3.3", default-features = false, features = ["std"], option
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"], optional = true }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"], optional = true }
 zstd = { version = "0.13.0", optional = true, default-features = false }
-zstd-sys = { version = "=2.0.9", optional = true, default-features = false } # TODO: temporary to fix parquet wasm build
-chrono = { workspace = true }
+# TODO: temporary to fix parquet wasm build
+# upstream issue: https://github.com/gyscos/zstd-rs/issues/269
+zstd-sys = { version = "=2.0.9", optional = true, default-features = false }chrono = { workspace = true }
 num = { version = "0.4", default-features = false }
 num-bigint = { version = "0.4", default-features = false }
 base64 = { version = "0.22", default-features = false, features = ["std", ], optional = true }
@@ -78,7 +79,9 @@ brotli = { version = "3.3", default-features = false, features = ["std"] }
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"] }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"] }
 zstd = { version = "0.13", default-features = false }
-zstd-sys = { version = "=2.0.9", default-features = false } # TODO: temporary to fix parquet wasm build
+# TODO: temporary to fix parquet wasm build
+# upstream issue: https://github.com/gyscos/zstd-rs/issues/269
+zstd-sys = { version = "=2.0.9", default-features = false }
 serde_json = { version = "1.0", features = ["std"], default-features = false }
 arrow = { workspace = true, features = ["ipc", "test_utils", "prettyprint", "json"] }
 tokio = { version = "1.0", default-features = false, features = ["macros", "rt", "io-util", "fs"] }

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -53,7 +53,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"], option
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"], optional = true }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"], optional = true }
 zstd = { version = "0.13.0", optional = true, default-features = false }
-zstd-sys = { version = "2.0.9", optional = true, default-features = false }
+zstd-sys = { version = "=2.0.9", optional = true, default-features = false }
 chrono = { workspace = true }
 num = { version = "0.4", default-features = false }
 num-bigint = { version = "0.4", default-features = false }
@@ -78,7 +78,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"] }
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"] }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"] }
 zstd = { version = "0.13", default-features = false }
-zstd-sys = { version = "2.0.9", default-features = false }
+zstd-sys = { version = "=2.0.9", default-features = false }
 serde_json = { version = "1.0", features = ["std"], default-features = false }
 arrow = { workspace = true, features = ["ipc", "test_utils", "prettyprint", "json"] }
 tokio = { version = "1.0", default-features = false, features = ["macros", "rt", "io-util", "fs"] }

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -53,7 +53,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"], option
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"], optional = true }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"], optional = true }
 zstd = { version = "0.13.0", optional = true, default-features = false }
-zstd-sys = { version = "=2.0.9", optional = true, default-features = false }
+zstd-sys = { version = "=2.0.9", optional = true, default-features = false } # TODO: temporary to fix parquet wasm build
 chrono = { workspace = true }
 num = { version = "0.4", default-features = false }
 num-bigint = { version = "0.4", default-features = false }
@@ -78,7 +78,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"] }
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"] }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"] }
 zstd = { version = "0.13", default-features = false }
-zstd-sys = { version = "=2.0.9", default-features = false }
+zstd-sys = { version = "=2.0.9", default-features = false } # TODO: temporary to fix parquet wasm build
 serde_json = { version = "1.0", features = ["std"], default-features = false }
 arrow = { workspace = true, features = ["ipc", "test_utils", "prettyprint", "json"] }
 tokio = { version = "1.0", default-features = false, features = ["macros", "rt", "io-util", "fs"] }
@@ -89,7 +89,7 @@ object_store = { version = "0.9.0", default-features = false, features = ["azure
 all-features = true
 
 [features]
-default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "zstd-sys", "base64"]
+default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "base64"]
 # Enable lz4
 lz4 = ["lz4_flex"]
 # Enable arrow reader/writer APIs
@@ -106,6 +106,8 @@ experimental = []
 async = ["futures", "tokio"]
 # Enable object_store integration
 object_store = ["dep:object_store", "async"]
+# Group Zstd dependencies
+zstd = ["dep:zstd", "zstd-sys"]
 
 [[example]]
 name = "read_parquet"

